### PR TITLE
Edit communication template

### LINKED ISSUE #445: Edit communication template component
This issue tracks the creation of an edit communication Template component.

### PR DESCRIPTION
**Pull request by jitsedesmet**
This PR aims to close #445 .

### CODE REVIEW

File: frontend/src/api/calls/communicationTemplateCalls.ts [hunk 1]
```diff
@@ -35,6 +41,17 @@ export async function createNewCommunicationTemplate(
     return <Promise<ICommunicationTemplate>>(await basePost(apiPaths.communicationTemplates, template)).data;
 }
 
+/**
+ * Function posting creating a new communication template on the backend.
```
Comment (andpuydt) [inline frontend/src/api/calls/communicationTemplateCalls.ts:45]:
```suggestion
 * Function posting the creation of a new communication template on the backend.
```

**Review by andpuydt — COMMENTED** [review]
Looks great! Especially the tests!

**Comment by jitsedesmet** [comment]
After all communication PR's are merged. The will be a cleanup PR.

**Review by andpuydt — DISMISSED** [review]

File: frontend/src/components/communication/createCommunicationTemplateForm.tsx [hunk 2]
```diff
@@ -1,58 +1,86 @@
...
-                        name="name"
-                        required
-                        placeHolder={capitalize(t("name"))}
-                        id="communicationTemplateNameField"
-                        data-testid="name"
-                    />
+                    <div>
+                        <label className="capitalize" htmlFor="communicationTemplateNameField">
+                            {capitalize(t("name")) + ":"}
+                        </label>
```
Comment (andpuydt) [inline frontend/src/components/communication/createCommunicationTemplateForm.tsx:45]:
```suggestion
                        <label htmlFor="communicationTemplateNameField">
                            {capitalize(t("name")) + ":"}
                        </label>
```

File: frontend/src/components/communication/createCommunicationTemplateForm.tsx [hunk 3]
```diff
@@ -1,58 +1,86 @@
...
+                                placeholder={capitalize(t("name"))}
+                                id="communicationTemplateNameField"
+                                data-testid="name"
+                            />
+                        )}
+                    </div>
+                    <div>
+                        <label className="capitalize" htmlFor="communicationTemplateSubjectField">
+                            {capitalize(t("subject")) + ":"}
+                        </label>
```
Comment (andpuydt) [inline frontend/src/components/communication/createCommunicationTemplateForm.tsx:62]:
```suggestion
                        <label htmlFor="communicationTemplateSubjectField">
                            {capitalize(t("subject")) + ":"}
                        </label>
```

File: frontend/src/pages/communicationTemplates/[id]/edit.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        return null;
+    }
+
+    const template: ICommunicationTemplate = receivedTemplate || emptyCommunicationTemplate;
+
+    return (
+        <div data-testid="communication-template-edit">
+            <h1 className="capitalize">
+                {capitalize(t("edit communicationTemplate")) + ": " + template.name}
+            </h1>
```
Comment (andpuydt) [inline frontend/src/pages/communicationTemplates/[id]/edit.tsx:34]:
```suggestion
            <h1>
                {capitalize(t("edit communicationTemplate")) + ": " + template.name}
            </h1>
```

**Review by andpuydt — DISMISSED** [review]

**Review by andpuydt — APPROVED** [review]

**Pull request merged**